tests/contrib/futures/test_propagation.py

```diff
@@ -154,7 +186,10 @@ def fn():
                 self.assertEqual(result, 42)
 
         # the trace must be completed
-        self.assert_structure(dict(name="executor.thread"))
```

Comment:
Why do are we removing self.assert_structure from tests? Is this method broken? Do we need to remove it in this PR?

Reply:
yes, because we are not propagating a context there isn't a strong link between the two.

this `assert_structure` stuff isn't very good/smart, because the `executor.thread` span doesn't have a `_parent` which points to the `main.thread` span, all of this stuff breaks.

we could do it in a separate PR, but the reason these assertions need to be refactored is because of the change being made in this PR.